src/models/Scholarships.ts

```diff
@@ -46,11 +46,15 @@ class Scholarships extends FirestoreCollection<ScholarshipData> {
    *
    * @param opts list options.
    */
-  list(opts: {
-    sortDir?: SortDirection;
-    sortField?: string;
-    authorId?: string;
-  }): Promise<FirestoreModel<ScholarshipData>[]> {
+  async list(
```

Comment:
No need to do `async`, this already returns a promise. The way you handle a promise that handles a promise is:

```js
return doSomething
  .then((result) => {/* do something with the result */});
```

See the [Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises#chaining) guide Promises